### PR TITLE
Feat: Add /export endpoint, To export All License Json

### PR DESCRIPTION
**Pull request by Kaushl2208**
image
image

### CODE REVIEW

File: pkg/api/licenses.go [hunk 1]
```diff
@@ -884,3 +884,61 @@ func SearchInLicense(c *gin.Context) {
...
+			Url:             license.Url,
+			AddDate:         license.AddDate.Format(time.RFC3339), // Convert time.Time to string
+			Copyleft:        fmt.Sprintf("%t", license.Copyleft),
+			FSFfree:         fmt.Sprintf("%t", license.FSFfree),
+			OSIapproved:     fmt.Sprintf("%t", license.OSIapproved),
+			GPLv2compatible: fmt.Sprintf("%t", license.GPLv2compatible),
+			GPLv3compatible: fmt.Sprintf("%t", license.GPLv3compatible),
+			Notes:           license.Notes,
+			Fedora:          license.Fedora,
+			TextUpdatable:   fmt.Sprintf("%t", license.TextUpdatable),
```
Comment (GMishx) [inline pkg/api/licenses.go:931]: `TextUpdatable` is a boolean field.

File: pkg/api/licenses.go [hunk 2]
```diff
@@ -884,3 +884,61 @@ func SearchInLicense(c *gin.Context) {
...
+			DetectorType:    license.DetectorType,
+			Active:          fmt.Sprintf("%t", license.Active),
```
Comment (GMishx) [inline pkg/api/licenses.go:933]: `Active` is a boolean field.

File: pkg/api/licenses.go [hunk 3]
```diff
@@ -884,3 +884,61 @@ func SearchInLicense(c *gin.Context) {
...
+			Source:          license.Source,
+			SpdxCompatible:  license.SpdxId, // Corrected field name to match your LicenseDB struct
```
Comment (GMishx) [inline pkg/api/licenses.go:935]: `SpdxCompatible` is a boolean field. `SpdxId` is a separate string field.

File: pkg/api/licenses.go [hunk 4]
```diff
@@ -884,3 +884,61 @@ func SearchInLicense(c *gin.Context) {
...
+			Risk:            fmt.Sprintf("%d", license.Risk),
+			Flag:            fmt.Sprintf("%d", license.Flag),
```
Comment (GMishx) [inline pkg/api/licenses.go:937]: `Risk` and `Flag` are integer fields.

File: pkg/api/licenses.go [hunk 5]
```diff
@@ -884,3 +884,61 @@ func SearchInLicense(c *gin.Context) {
...
+			Marydone:        fmt.Sprintf("%t", license.Marydone),
```
Comment (GMishx) [inline pkg/api/licenses.go:938]: `Marydone` is a boolean field.

File: pkg/api/licenses.go [hunk 6]
```diff
@@ -884,3 +884,61 @@ func SearchInLicense(c *gin.Context) {
...
+		}
+		licenseJson = append(licenseJson, jsonLicense)
```
Comment (GMishx) [inline pkg/api/licenses.go:940]: Not sure but won't doing so won't make `licenseJson` as a nested list?

**Review by GMishx — CHANGES_REQUESTED** [review]

File: pkg/api/licenses.go (hunk 6 above)
Comment (avinal) [reply pkg/api/licenses.go:940]: Do we want something like?

```json
{
    "licenses": [
       {...},
       {...}
     ]
}
```

or 

```json
[
    {...},
    {...}
]
```

the above will create the second one.

File: pkg/api/licenses.go [hunk 7]
```diff
@@ -884,3 +884,61 @@ func SearchInLicense(c *gin.Context) {
...
+//
+//	@Summary		Export all licenses as a json file
+//	@Description	Export all licenses as a json file
+//	@Id				ExportLicenses
+//	@Tags			Licenses
+//	@Accept			json
+//	@Produce		json
+//	@Success		200	{array}		models.LicenseJson
+//	@Failure		500	{object}	models.LicenseError	"Failed to fetch obligations"
+//	@Security		ApiKeyAuth
```
Comment (avinal) [inline pkg/api/licenses.go:898]:
```suggestion
//	@Failure		500	{object}	models.LicenseError	"Failed to fetch licenses"
```
It should be licenses and the endpoint in unauthorized.

File: pkg/api/licenses.go [hunk 8]
```diff
@@ -884,3 +884,61 @@ func SearchInLicense(c *gin.Context) {
...
 	c.JSON(http.StatusOK, res)
 }
+
+// ExportLicenses gives users all licenses as a json file.
+//
+//	@Summary		Export all licenses as a json file
+//	@Description	Export all licenses as a json file
+//	@Id				ExportLicenses
+//	@Tags			Licenses
+//	@Accept			json
```
Comment (avinal) [inline pkg/api/licenses.go:894]:
```suggestion
```
Is this needed, since this endpoint does not accept anything?

**Review by avinal — CHANGES_REQUESTED** [review]
Few changes, AFIR the docs are generated so regenerating after changes will fix them as well.

File: pkg/api/api.go [hunk 9]
```diff
@@ -71,6 +71,7 @@ func Router() *gin.Engine {
 		{
 			licenses.GET("", FilterLicense)
 			licenses.GET(":shortname", GetLicense)
+			licenses.GET("/export", ExportLicenses)
```
Comment (avinal) [inline pkg/api/api.go:74]:
```suggestion
			licenses.GET("export", ExportLicenses)
```

**Review by GMishx — APPROVED** [review]
Changes looks good.

**Comment by GMishx** [comment]
@Kaushl2208 please squash the commits and sign them with you keys.

File: pkg/api/licenses.go [hunk 10]
```diff
@@ -884,3 +884,34 @@ func SearchInLicense(c *gin.Context) {
...
 }
+
+// ExportLicenses gives users all licenses as a json file.
+//
+//	@Summary		Export all licenses as a json file
+//	@Description	Export all licenses as a json file
+//	@Id				ExportLicenses
+//	@Tags			Licenses
+//	@Produce		json
+//	@Success		200	{array}		models.LicenseJson
```
Comment (GMishx) [inline pkg/api/licenses.go:895]: Oops, a minor change needed.
```suggestion
//	@Success		200	{array}		models.ExportLicenseDB
```

**Review by GMishx — CHANGES_REQUESTED** [review]

File: pkg/api/licenses.go (hunk 10 above)
Comment (GMishx) [reply pkg/api/licenses.go:895]: @Kaushl2208 , Can you please change the file name with following comment as well?

https://github.com/fossology/LicenseDb/pull/53#issuecomment-2109355660

**Comment by Kaushl2208** [comment]
image

Changed the filename :)

**Review by GMishx — APPROVED** [review]
LGTM

**Review by GMishx — APPROVED** [review]
Tested, working as expected.

**Comment by GMishx** [comment]
@Kaushl2208 can you please fix the commit message and sign the commits so we can merge your PR?

[Learn more about signing commits.](https://docs.github.com/authentication/managing-commit-signature-verification/about-commit-signature-verification)

**Pull request merged**